### PR TITLE
python3Packages.schemdraw: init at 0.16, add dependencies

### PR DESCRIPTION
**Pull request by SFrijters**
###### Description of changes

Primary goal is to add `schemdraw`, but it needs some more dependencies that are not yet available in nixpkgs...

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SFrijters** [comment]
Result of `nixpkgs-review pr 222012` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>latex2mathml (python310Packages.latex2mathml)</li>
    <li>latex2mathml.dist (python310Packages.latex2mathml.dist)</li>
    <li>python310Packages.schemdraw</li>
    <li>python310Packages.schemdraw.dist</li>
    <li>python310Packages.ziafont</li>
    <li>python310Packages.ziafont.dist</li>
    <li>python310Packages.ziamath</li>
    <li>python310Packages.ziamath.dist</li>
    <li>python311Packages.latex2mathml</li>
    <li>python311Packages.latex2mathml.dist</li>
    <li>python311Packages.schemdraw</li>
    <li>python311Packages.schemdraw.dist</li>
    <li>python311Packages.ziafont</li>
    <li>python311Packages.ziafont.dist</li>
    <li>python311Packages.ziamath</li>
    <li>python311Packages.ziamath.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/ziamath/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    rev = "${version}";
+    hash = "sha256-JuuCDww0EZEHZLxB5oQrWEJpv0szjwe4iXCRGl7OYTA=";
+  };
+
+  propagatedBuildInputs = [
+    ziafont
+  ];
+
+  nativeCheckInputs = [
+    pytest
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ziamath/default.nix:29]:
```suggestion
    pytestCheckHook
```

File: pkgs/development/python-modules/ziamath/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    nbval
+    latex2mathml
+  ];
+
+  checkPhase = ''
+    py.test --nbval-lax test
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ziamath/default.nix:35]: Please use pytestFlagsArray to supply nbval-lax

File: pkgs/development/python-modules/ziamath/default.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+buildPythonPackage rec {
+  pname = "ziamath";
+  version = "0.7";
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromBitbucket {
+    owner = "cdelker";
+    repo = pname;
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ziamath/default.nix:20]:
```suggestion
    rev = version;
```

File: pkgs/development/python-modules/ziafont/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  };
+
+  nativeCheckInputs = [
+    pytest
+    nbval
+  ];
+
+  checkPhase = ''
+    rm test/manyfonts.ipynb  # Tries to download fonts
+    py.test --nbval-lax test
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ziafont/default.nix:29]: Please use pytestCheckHook here, too

File: pkgs/development/python-modules/ziafont/default.nix [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+buildPythonPackage rec {
+  pname = "ziafont";
+  version = "0.5";
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromBitbucket {
+    owner = "cdelker";
+    repo = pname;
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ziafont/default.nix:18]:
```suggestion
    rev = version;
```

File: pkgs/development/python-modules/schemdraw/default.nix [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    ziamath
+  ];
+
+  # Strip out references to unfree fonts from the test suite
+  postPatch = ''
+    substituteInPlace test/test_styles.ipynb --replace "font='Times', " ""
+  '';
+
+  checkPhase = ''
+    sh ./runtest.sh
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/schemdraw/default.nix:60]: Please do not use peoples wrappers scripts. We want to execute pytest through our hook

File: pkgs/development/python-modules/schemdraw/default.nix [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    svgmath = [
+      latex2mathml
+      ziafont
+      ziamath
+    ];
+  };
+
+  nativeCheckInputs = [
+    pytest
+    pytest-cov
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/schemdraw/default.nix:46]:
```suggestion
```
collecting coverage just slows down tests

File: pkgs/development/python-modules/latex2mathml/default.nix [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    poetry-core
+  ];
+
+  propagatedBuildInputs = [
+    setuptools  # needs pkg_resources at runtime
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pytest-cov
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/latex2mathml/default.nix:38]:
```suggestion
```

File: pkgs/development/python-modules/latex2mathml/default.nix [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+buildPythonPackage rec {
+  pname = "latex2mathml";
+  version = "3.75.2";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "roniemartinez";
+    repo = pname;
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/latex2mathml/default.nix:22]:
```suggestion
    rev = version;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/latex2mathml/default.nix (hunk 8 above)
Comment (SFrijters) [reply pkgs/development/python-modules/latex2mathml/default.nix:38]: This is a bit of a problem: pyproject.toml explicitly sets coverage related flags and the build fails if I don't provide pytest-cov. How do you suggest I patch that out? (Or do we just keep pytest-cov after all?)

Comment (SuperSandro2000) [reply pkgs/development/python-modules/latex2mathml/default.nix:38]: > How do you suggest I patch that out?

Yes

**Comment by SFrijters** [comment]
Result of `nixpkgs-review pr 222012` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>latex2mathml (python310Packages.latex2mathml)</li>
    <li>latex2mathml.dist (python310Packages.latex2mathml.dist)</li>
    <li>python310Packages.schemdraw</li>
    <li>python310Packages.schemdraw.dist</li>
    <li>python310Packages.ziafont</li>
    <li>python310Packages.ziafont.dist</li>
    <li>python310Packages.ziamath</li>
    <li>python310Packages.ziamath.dist</li>
    <li>python311Packages.latex2mathml</li>
    <li>python311Packages.latex2mathml.dist</li>
    <li>python311Packages.schemdraw</li>
    <li>python311Packages.schemdraw.dist</li>
    <li>python311Packages.ziafont</li>
    <li>python311Packages.ziafont.dist</li>
    <li>python311Packages.ziamath</li>
    <li>python311Packages.ziamath.dist</li>
  </ul>
</details>

**Pull request merged**